web/src/types.ts

```diff
@@ -42,3 +42,21 @@ export type SearchResult = {
   result: Entry;
 };
 export type CurrentSearch = (AssistantEntry & { phase?: string }) | undefined;
+
+export type Mode = "rookie" | "concise" | "default";
+
+export type LLMSettings = {
+  prompts?: {
```

Comment:
Are you sure all these settings should have the option of being undefined?

Reply:
yes. They should have sane defaults in the API - they're mainly here to allow playing about with them, so no point in enforcing them all to be set